### PR TITLE
Boost: Implement "Clear Cache" UI

### LINKED ISSUE #35783: [Boost] Implement "Clear Cache" UI
Currently static.

### PR DESCRIPTION
**Pull request by dilirity**
Fixes #35783

## Proposed changes:

* Clear page cache when clicking the "Clear Cache" button;
* Disable "Clear Cache" button during cleanup;
* Update the Page Cache summary during cleanup;
* Add snackbar message after cleanup is done.

Currently, when cleaning the cache, the "clean cache" PHP side of things returns an error that the cache directory doesn't exist. This is why the snackbar message shows on success and error.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

pc9hqz-2rA-p2

pc9hqz-2vF-p2

## Does this pull request change what data or activity we track or use?

no

## Testing instructions:

* Make sure you have some cached pages;
* Click the "Clear Cache" button and make sure cache is deleted (and that the message shows up).

### CODE REVIEW

File: projects/plugins/boost/app/assets/src/js/lib/stores/page-cache.ts [hunk 1]
```diff
@@ -28,6 +31,13 @@ export function useRunPageCacheSetupAction() {
 	return usePageCacheErrorAction( 'run-page-cache-setup', z.void() );
 }
 
+/**
+ * Hook which creates a callable action for clearing Page Cache.
+ */
+export function useClearPageCacheAction() {
+	return usePageCacheAction( 'clear-page-cache', z.void() );
+}
```
Comment (pyronaur) [inline projects/plugins/boost/app/assets/src/js/lib/stores/page-cache.ts]: I think `usePageCacheAction` could probably be merged with this

**Review by pyronaur — DISMISSED** [review]
Looks good, left a small suggestion.

**Review by donnchawp — DISMISSED** [review]
Worked for me. UI showed as expected and cache was cleared.

**Review by donnchawp — APPROVED** [review]
Looks great!

**Pull request merged**